src/main/java/com/kodilla/ecommercee/domain/Order.java

```diff
@@ -38,11 +39,23 @@
     @Column(name = "SHIPPINGADDRESS")
     private String shippingAddress;
 
-    @ManyToOne
+
+    @ManyToOne(fetch = FetchType.LAZY)
     @JoinColumn(name = "user_id", nullable = false)
     private User user;
 
     @OneToOne(fetch = FetchType.LAZY)
     @JoinColumn(name = "cart_id", nullable = false , referencedColumnName = "ID")
     private Cart cart;
+
+
+    @JsonBackReference
+    public User getUser() {
+        return user;
+    }
+
+    @JsonBackReference
+    public Cart getCart() {
+        return cart;
+    }
 }
```

Comment:
W nawiązaniu do wcześniejszego komentarza - dzięki ułożeniu struktury Dto w odpowiedni sposób nie będzie trzeba tworzyć osobnych konstruktorów i getterów, które nadpisują adnotacje lomboka.
@danieloginski mi kiedyś powiedział, że zamiast tworzyć alternatywne konstruktory w encji - powinno się tworzyć osobne klasy DTO dla konkretnego zastosowania. Nie wiem czy to tutaj ma zastosowanie, rozmawiałem też z @wezik  na ten temat w kwestii naszych klas i myślimy tutaj podobnie. Daniel ?

Reply:
pewnie, jak działa to czemu nie skorzystać , podmienię to ASAP , dzięki